cdap-api/src/main/java/co/cask/cdap/api/dataset/table/RowColumns.java

```diff
@@ -69,10 +71,40 @@ public RowColumns(String row, String... columns) {
   }
 
   @SuppressWarnings("unchecked")
-  public T add(String... columns) {
+  public T add(String firstColumn, String... moreColumns) {
```

Comment:
as I already mentioned, this is really awkward if you have code that dynamically computes the set of columns.